### PR TITLE
Created LetterCombinationsOfPhoneNumberTest.java

### PR DESCRIPTION
**Pull request by TaranjeetSinghKalsi**
Renamed List_all_Possible_Words_From_Phone_Digits.java  to LetterCombinationsOfPhoneNumber.java and Created LetterCombinationsOfPhoneNumberTest.java for JUnit Tests. Also changed the access specifier of functions in LetterCombinationsOfPhoneNumber.java from private to protected for JUnit Tests.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Test
+    public void letterCombinationsOfPhoneNumber() {
+        LetterCombinationsOfPhoneNumber ob = new LetterCombinationsOfPhoneNumber();
+        ob.generateNumberToCharMap();
+
+        // ** Test 1 **
+        // Input: digits = ""
+        // Output: []
+        int numbers1[] = {};
```
Comment (yanglbme) [inline src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java:17]:
```suggestion
        int[] numbers1 = {};
```

File: src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        List<String> output1 = Arrays.asList("");
+        assertTrue(ob.printWords(numbers1, numbers1.length, 0, "").equals(output1));
+
+        // ** Test 2 **
+        // Input: digits = "2"
+        // Output: ["a","b","c"]
+        int numbers2[] = { 2 };
```
Comment (yanglbme) [inline src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java:24]:
```suggestion
        int[] numbers2 = { 2 };
```

File: src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        List<String> output2 = Arrays.asList("a", "b", "c");
+        assertTrue(ob.printWords(numbers2, numbers2.length, 0, "").equals(output2));
+
+        // ** Test 3 **
+        // Input: digits = "23"
+        // Output: ["ad","ae","af","bd","be","bf","cd","ce","cf"]
+        int numbers3[] = { 2, 3 };
```
Comment (yanglbme) [inline src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java:31]:
```suggestion
        int[] numbers3 = { 2, 3 };
```

File: src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        List<String> output3 = Arrays.asList("ad", "ae", "af", "bd", "be", "bf", "cd", "ce", "cf");
+        assertTrue(ob.printWords(numbers3, numbers3.length, 0, "").equals(output3));
+
+        // ** Test 4 **
+        // Input: digits = "234"
+        // Output: ["adg", "adh", "adi", "aeg", "aeh", "aei", "afg", "afh", "afi",
+        // "bdg", "bdh", "bdi", "beg", "beh", "bei", "bfg", "bfh", "bfi", "cdg", "cdh",
+        // "cdi", "ceg", "ceh", "cei", "cfg", "cfh", "cfi"]
+        int numbers4[] = { 2, 3, 4 };
```
Comment (yanglbme) [inline src/test/java/com/thealgorithms/strings/LetterCombinationsOfPhoneNumberTest.java:40]:
```suggestion
        int[] numbers4 = { 2, 3, 4 };
```

**Review by yanglbme — CHANGES_REQUESTED** [review]
Move `[]` to the variable type would be better :)

**Comment by TaranjeetSinghKalsi** [comment]
> Move `[]` to the variable type would be better :)
Done

**Review by TaranjeetSinghKalsi — COMMENTED** [review]
Kindly approve it

**Review by TaranjeetSinghKalsi — COMMENTED** [review]
Done

**Review by yanglbme — APPROVED** [review]

**Pull request merged**